biomaj/bank.py

```diff
@@ -884,60 +883,71 @@ def removeAll(self, force=False):
         return True
 
     def get_status(self):
-        '''
+        """
         Get status of current workflow
 
         :return: dict of current workflow status
-        '''
+        """
         if self.bank['status'] is None:
             return {}
         return self.bank['status']
 
-    def remove_pending(self, release):
-        '''
-        Remove pending releases
+    def remove_pending(self, release=None):
+        """
+        Remove pending releases if 'release is None
 
-        :param release: release or release directory
+        :param release: release or release directory, default None
         :type release: str
         :return: bool
-        '''
-        release = str(release)
+        """
+        if release:
+            release = str(release)
         logging.warning('Bank:' + self.name + ':RemovePending')
 
         if not self.is_owner():
             logging.error('Not authorized, bank owned by ' + self.bank['properties']['owner'])
             raise Exception('Not authorized, bank owned by ' + self.bank['properties']['owner'])
 
-        if not self.bank['pending']:
+        if 'pending' not in self.bank:
             return True
         pendings = self.bank['pending']
-        for release in list(pendings.keys()):
-            pending_session_id = pendings[release]
+
+        for pending in pendings:
+            # Only work with pending for argument release
+            if release and release != pending['release']:
+                continue
+            pending_session_id = pending['id']
             pending_session = None
             for s in self.bank['sessions']:
                 if s['id'] == pending_session_id:
                     pending_session = s
                     break
             session = Session(self.name, self.config, RemoveWorkflow.FLOW)
             if pending_session is None:
-                session._session['release'] = release
+                session._session['release'] = pending['release']
             else:
                 session.load(pending_session)
             if os.path.exists(session.get_full_release_directory()):
                 logging.debug("Remove:Pending:Dir:" + session.get_full_release_directory())
                 shutil.rmtree(session.get_full_release_directory())
-            self.remove_session(pendings[release])
-        self.banks.update({'name': self.name}, {'$set': {'pending': {}}})
+            self.remove_session(release['id'])
+            # Only remove pending session for this release
+            if release:
```

Comment:
this is not needed, session will be pulled from pending in remove_session method call.